### PR TITLE
Update service client doc comments

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java [hunk 1]
```diff
@@ -38,7 +38,8 @@
 import java.util.concurrent.Executors;
 
 /**
- * Use the RegistryManager class to manage the identity registry in IoT Hubs.
+ * Use the RegistryManager client to manage the identity registry in IoT hubs.
```
Comment (drwill-ms) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java:41]: I'd have preferred all our clients end with the suffix "Client", but we can at least refer to them as clients.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java [hunk 2]
```diff
@@ -19,7 +19,8 @@
 import java.util.concurrent.Executors;
 
 /**
- * Extend the ServiceClient class and provide AMPQ specific implementation.
+ * Use the ServiceClient to send and monitor messages to devices in IoT hubs.
```
Comment (drwill-ms) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java:22]: I updated all the clients to use the same wording.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java [hunk 3]
```diff
@@ -38,7 +38,8 @@
...
+ * To access twins, use the {@link com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin} client.
```
Comment (drwill-ms) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java:42]: Added a link to a related client.

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java [hunk 4]
```diff
@@ -19,7 +19,8 @@
 import java.util.concurrent.Executors;
 
 /**
- * Extend the ServiceClient class and provide AMPQ specific implementation.
```
Comment (drwill-ms) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java:22]: I couldn't make heads or tails of this doc comment. Any idea what we were trying to say?

Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java:22]: This sounds like copy-paste error from the .NET SDK, back when the .NET SDK had a ServiceClient class and an AmqpServiceClient class. I wouldn't worry about deleting it. This java SDK never had that same pattern

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceMethod.java [hunk 5]
```diff
@@ -25,7 +25,7 @@
 import java.util.Objects;
 
 /**
- * DeviceMethod enables service client to directly invoke methods on various devices from service client.
+ * Use the DeviceMethod client to directly invoke methods on various devices in IoT hub.
```
Comment (timtay-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceMethod.java:28]: devices **and modules**

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**